remote/src/main/java/com/spotify/metrics/remote/LimitedRemote.java

```diff
@@ -23,23 +23,30 @@
 
 import com.google.common.util.concurrent.ListenableFuture;
 import com.spotify.futures.ConcurrencyLimiter;
-
 import java.util.Map;
 import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
 /**
  * A remote that wraps another remote and limits both the number of
  * requests in flight as well as the number of requests in queue.
  */
 public class LimitedRemote implements Remote {
+    private static final Executor executorService = Executors.newSingleThreadExecutor();
 
     private final Remote inner;
     private final ConcurrencyLimiter<Integer> tasks;
 
     public LimitedRemote(Remote inner, int maxConcurrency, int highWaterMark) {
         this.inner = inner;
-        tasks = ConcurrencyLimiter.create(maxConcurrency, highWaterMark);
+        tasks = ConcurrencyLimiter.create(executorService, maxConcurrency, highWaterMark);
```

Comment:
You can call the overloaded ctor here instead. 
The default Executor was the [directExecutor](https://github.com/spotify/futures-extra/commit/d79facc938eaa1e2d62f6235ffc0b2eae392dd1f#diff-f0b70a75659f233b0a6e876ccabef30a) if you wan't to keep that.